### ISSUE TITLE
Package issues

### ISSUE THREAD

**Issue by bweedop** [pearselab/r-world-bweedop#3]
I feel like this should be simple but I am missing something. My plants.r is calling for the 'terrain' matrix that originates from my terrain.r. When I do R CMD check, I get a note telling me that 'no visible binding for global variable: terrain' 
How do I go about solving the problem of having an object on one .r and then using it on another .r  without causing problems with the check?

**Comment by willpearse** [comment]
1. No variables in a package, only functions. It sounds like you are doing
   this.
2. Make sure you have used @export on all your functions

Sorry to be brief it's 6pm I'm on the train

**Comment by bweedop** [comment]
I don't think I have any variables in the package. The issue is that I have my run.eco function using the terrain matrix:
`run.eco<-function(timesteps, ...){
  plants <- array("", dim=c(dim(terra),timesteps+1))
  for(i in seq_len(dim(plants)[3]))
    plants[,,i][is.na(terra)] <- NA
  for(k in (seq_len(dim(plants)[3]-1)))
    plants[,,k] <- plant.timestep(plants[,,k], info)
  k<-k+1
  return(plants)
}`
and R CMD check checks my example:
plants<-run.eco(10)

The check gives me an error saying: object 'terra' not found

**Comment by willpearse** [comment]
Right, I see. There is a bug in this function. You're using terra - is it
an argument to your function?
On Fri, 18 Nov 2016 at 18:14, bweedop notifications@github.com wrote:

I don't think I have any variables in the package. The issue is that I have
my run.eco function using the terrain matrix:
run.eco<-function(timesteps, ...){
plants <- array("", dim=c(dim(terra),timesteps+1))
for(i in seq_len(dim(plants)[3]))
plants[,,i][is.na(terra)] <- NA
for(k in (seq_len(dim(plants)[3]-1)))
plants[,,k] <- plant.timestep(plants[,,k], info)
k<-k+1
return(plants)
}
and R CMD check checks my example:
plants<-run.eco(10)

The check gives me an error saying: object 'terra' not found

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
https://github.com/pearselab/r-world-bweedop/issues/3#issuecomment-261601492,
or mute the thread

**Comment by bweedop** [comment]
Right now it is not but I tried using it as an argument and it gave me the same error...
